src/main/java/com/jpop/productservice/controller/ProductServiceController.java

```diff
@@ -0,0 +1,68 @@
+package com.jpop.productservice.controller;
+
+import com.google.gson.Gson;
+import com.jpop.productservice.model.Product;
+import com.jpop.productservice.service.ProductDeleteService;
+import com.jpop.productservice.service.ProductDetailService;
+import com.jpop.productservice.service.ProductInsertService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/api/v1/products")
+public class ProductServiceController {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProductServiceController.class);
+    private ProductInsertService productInsertService;
+    private ProductDetailService productDetailService;
+    private ProductDeleteService productDeleteService;
+
+    @Autowired
+    public ProductServiceController(ProductInsertService productInsertService, ProductDetailService productDetailService, ProductDeleteService productDeleteService) {
+        this.productInsertService = productInsertService;
+        this.productDetailService = productDetailService;
+        this.productDeleteService = productDeleteService;
+    }
+
+    @GetMapping(value = "/", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<Product>> getAllProducts() {
+        return ResponseEntity.ok(productDetailService.getAvailableProducts());
+    }
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> insertProduct(@RequestBody Product payload) {
+
+        logger.info("Payload received to insert data in product service with value {}", payload);
+        productInsertService.processRawData(payload);
+        return ResponseEntity.status(HttpStatus.CREATED).body("Successfully added product in repository");
```

Comment:
It is suggested to send the URI (location) in the header back to the user to access the created resource. Try to use `ResponseEntity.created(uri of the resource).body(product object which is created)`. The intention is that the client of this endpoint has to know the id of the product inserted.

Reply:
Thank you so much for the suggestion on this.